Update "Data Sources" page in documentation

Steph-Chong opened the pull request:
This PR updates the "Data Sources" page in the documentation (data.md).

As I don't have a met background, I thought it might be prudent for somebody with a met background to check my work. @nicholasloveday @BethEbert - if either or both of you are happy to take a look, that would be much appreciated. 

You can see the updated Data Sources page here: https://scores.readthedocs.io/en/228-documentation-testing-branch/data.html 

(quick note from Tennessee - the link above will also now show an alert that you may be reading an old version of the documentation. In this case, just close the box, ignore the message, and do the review on the linked version)

Changes made:
- I added that `scores` is used with xarray and pandas, as that was not mentioned previously on this page.
- I tried to retain the existing information, while improving clarity.
- I added new headers and header levels, and reorganised some of the information
- I made all existing URLs hyperlinks
- I added some links
- I updated some links that no longer worked, replaced one dataset link with a DOI (that resolves to the same page as the initial link), and replaced the link for WeatherBench to WeatherBench 2 (as far as I can tell, WeatherBench 2 still includes ERA 5 and supplants WeatherBench - but I didn't compare the two in detail, so I cannot guarantee this).
- I modified the capitalisation of headers for improved consistency with the rest of the documentation. 

I also have two questions that I have included as code comments (either see below or "Files changed")

Steph-Chong commented on docs/data.md:
```diff
@@ -1,40 +1,59 @@
-Radar data also varies according to region and is not a globally standardised data set. Information on Australian based radars can be found at https://www.openradar.io/
+#### Point-Based Data
+
+Point-based observations are shared routinely between countries for the purposes of weather modelling.
```
I couldn't work out whether this sentence should be retained, and if so, where in the paragraph it should go.

Steph-Chong commented on docs/data.md:
```diff
@@ -1,40 +1,59 @@
+#### CliMetLab
+
+The European Centre for Medium-Range Weather Forecasts (ECMWF) has developed the CliMetLab Python package to simplify access to a large range of climatological and meteorological datasets. See [https://climetlab.readthedocs.io/](https://climetlab.readthedocs.io/).
+
```
I also wasn't sure whether WeatherBench 2 should also be added to the "Software for Accessing Data" section.

Steph-Chong commented on docs/data.md:
```diff
@@ -1,40 +1,59 @@
-Radar data also varies according to region and is not a globally standardised data set. Information on Australian based radars can be found at https://www.openradar.io/
+#### Point-Based Data
+
+Point-based observations are shared routinely between countries for the purposes of weather modelling.
+
```
I couldn't work out whether this sentence should be retained, and if so, where in the section it should go

Steph-Chong commented on docs/data.md:
```diff
@@ -1,40 +1,59 @@
+
+Radar data varies according to region and is not a globally standardised data set. Information on Australian based radars can be found at [https://www.openradar.io/](https://www.openradar.io/).
+
+### Software for Acessing Data
+
```
I wasn't sure whether WeatherBench 2 should also be added as an entry to the "Software for Accessing Data" section.

tennlee commented:
I have read through it and think it's a good change, and I'm happy with the page as-is. Getting a few different perspectives on this one makes sense to me to make sure it can be easily understood be several people and also makes sense for those in the field.

Steph-Chong commented on docs/data.md:
```diff
@@ -1,40 +1,60 @@
+
+Radar data varies according to region and is not a globally standardised data set. Information on Australian based radars can be found at [https://www.openradar.io/](https://www.openradar.io/).
+
+### Software for Accessing Data
```
I wasn't sure whether WeatherBench 2 should also be added as an entry to the "Software for Accessing Data" section.

Steph-Chong commented on docs/data.md:
```diff
@@ -1,40 +1,60 @@
+
+#### Point-Based Data
+
+Point-based observations are shared routinely between countries for the purposes of weather modelling.
+
```
I wasn't sure whether this sentence should be retained, and if so, I wasn't sure where in the section it should go.

Steph-Chong replied:
This sentence has since been modified, so this is now resolved.

Steph-Chong commented on docs/data.md:
```diff
@@ -1,40 +1,60 @@
+Radar data varies according to region and is not a globally standardised data set. Information on Australian based radars can be found at [https://www.openradar.io/](https://www.openradar.io/).
+
+### Software for Accessing Data
+
```
I wasn't sure if WeatherBench 2 should also be added to this section

Steph-Chong commented on docs/data.md:
```diff
@@ -1,40 +1,60 @@
+
+Radar data varies according to region and is not a globally standardised data set. Information on Australian based radars can be found at [https://www.openradar.io/](https://www.openradar.io/).
+
+### Software for Accessing Data
+
```
Should WeatherBench 2 also be added to this section?

Steph-Chong replied:
Tennessee said (offline) to add WeatherBench 2 to this section. I have now done so.

BethEbert commented:
Hi Steph and Tennessee,

I took the easy (for me) option and put some comments into the PDF version you sent before making changes. I think you may have fixed a few typos, but there are a few other comments to consider.

The data page is looking good!

Beth
________________________________
From: Stephanie Chong ***@***.***>
Sent: Sunday, 16 June 2024 15:23
To: nci/scores ***@***.***>
Cc: Beth Ebert ***@***.***>; Mention ***@***.***>
Subject: Re: [nci/scores] Update "Data Sources" page in documentation (PR #544)


@Steph-Chong commented on this pull request.

________________________________

In docs/data.md<https://github.com/nci/scores/pull/544#discussion_r1641616586>:

> +### Software for Accessing Data
+


Should WeatherBench 2 also be added to this section?

—
Reply to this email directly, view it on GitHub<https://github.com/nci/scores/pull/544#pullrequestreview-2121091267>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AGMQHAPBXXET7SACNC3RMATZHUOOJAVCNFSM6AAAAABJLPTMPKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDCMRRGA4TCMRWG4>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

Steph-Chong commented on docs/data.md:
```diff
@@ -1,40 +1,60 @@
 
-For more information see https://www.ecmwf.int/en/forecasts/datasets/reanalysis-datasets/era5 and
-https://github.com/pangeo-data/WeatherBench
+To use `scores` with [NetCDF](https://doi.org/10.5065/D6H70CW6) or [HDF5](https://doi.org/10.11578/dc.20180330.1) data, install [h5netcdf](https://github.com/h5netcdf/h5netcdf). The h5netcdf library is included in the `scores` ["all"](installation.md#all-dependencies-excludes-some-maintainer-only-packages) and ["tutorial"](installation.md#tutorial-dependencies) installation options. Opening NetCD ...
```
On further reflection, the HDF5 link I included was the one used in the JOSS draft (and is probably fine for that purpose). However, it points to a specific release, which doesn't appear to be the latest release. Rather than needing to stay abreast of the HDF5 release cycle, my guess is for this specific context, it would be better to replace the HDF5 link with something else - perhaps their GitHub repo?

tennlee replied:
I think on balance the github link is the most likely to be the most useful link for users of this page

Steph-Chong replied:
I have updated to the GitHub link

Steph-Chong commented:
> Hi Steph and Tennessee, I took the easy (for me) option and put some comments into the PDF version you sent before making changes. I think you may have fixed a few typos, but there are a few other comments to consider. The data page is looking good! Beth

Thanks very much @BethEbert. Unfortunately neither Tennessee nor I can find the PDF you added comments to - so we can't see your feedback.

Tennessee suggested if you are happy to, you could either email him your feedback, or you could post your feedback as comments here on GitHub.

When I looked at the page again earlier today, as well as fixing a few typos, I reordered the sections (again) and also added a sentence in one section - so there were a few further changes I made earlier today. 

This PDF shows the page in its current form: 
[Data Sources 16.6.24.pdf](https://github.com/user-attachments/files/15856499/Data.Sources.16.6.24.pdf)

BethEbert commented:
Hi again Steph and Tennessee,

Not sure why the PDF attachment didn't come through. Here it is again - hopefully this time successful. Please don't feel obligated to use any of it.

Beth

________________________________
From: Stephanie Chong ***@***.******@***.***>>
Sent: Sunday, 16 June 2024 17:23
To: nci/scores ***@***.******@***.***>>
Cc: Beth Ebert ***@***.******@***.***>>; Mention ***@***.******@***.***>>
Subject: Re: [nci/scores] Update "Data Sources" page in documentation (PR #544)


Hi Steph and Tennessee, I took the easy (for me) option and put some comments into the PDF version you sent before making changes. I think you may have fixed a few typos, but there are a few other comments to consider. The data page is looking good! Beth

Thanks very much @BethEbert<https://github.com/BethEbert>. Unfortunately neither Tennessee nor I can find the PDF you added comments to - so we can't see your feedback.

Tennessee suggested if you are happy to, you could either email him your feedback, or you could post your feedback as comments here on GitHub.

When I looked at the page again earlier today, as well as fixing a few typos, I reordered the sections (again) and also added a sentence in one section - so there were a few further changes I made earlier today.

This PDF shows the page in its current form:
Data Sources 16.6.24.pdf<https://github.com/user-attachments/files/15856499/Data.Sources.16.6.24.pdf>

-
Reply to this email directly, view it on GitHub<https://github.com/nci/scores/pull/544#issuecomment-2171124593>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AGMQHAOUQWAKBDGLW62LGNLZHU4QBAVCNFSM6AAAAABJLPTMPKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNZRGEZDINJZGM>.
You are receiving this because you were mentioned.

Steph-Chong commented:
> Hi again Steph and Tennessee, Not sure why the PDF attachment didn't come through. Here it is again - hopefully this time successful. Please don't feel obligated to use any of it. Beth

Thanks for the helpful feedback Beth. When I get a moment, I will update the page to incorporate your feedback.

Steph-Chong commented:
I have made further updates to:
- Incorporate feedback from Beth
- Change HDF5 link to their GitHub
- Add WeatherBench 2 to the software packages section
- Changed one "data set" to "dataset" for consistency with the rest of the page.

When I built it locally, it rendered correctly in readthedocs.

I intend to read through it again later, as I find I sometimes pick up missed typos etc. when I reread my work after a break. But if anyone is keen to merge it before then, please feel free to do a check for any missed typos etc. (And of course, if anyone notices any mistakes, please do point them out).

Steph-Chong commented:
There are two areas where I would particularly appreciate feedback:

1. WeatherBench 2: as I don't have a background in this area, it would be much appreciated if someone could please double check what I have written is accurate and/or if there is anything else that should be mentioned in this section.

2. The "Point-Based Data" subsection: this is the only sub-section in the "Datasets" section that doesn't include a brief description of what point-based observations are. I am guessing point-based observations are exactly what they sound like (observations taken an specific locations), so perhaps an explanation is unnecessary. But I wanted to raise it, in case (a) it would be preferable to provide a one sentence description and (b) if so, it would be helpful if someone could suggest a suitable sentence (otherwise I can try and come up with one, and seek feedback).

nicholasloveday commented:
> There are two areas where I would particularly appreciate feedback:
> 
> 1. WeatherBench 2: as I don't have a background in this area, it would be much appreciated if someone could please double check what I have written is accurate and/or if there is anything else that should be mentioned in this section.
> 2. The "Point-Based Data" subsection: doesn't include a brief description of what point-based observations are. (The other "Datasets" subsections provide a brief description of what type of data is included in the dataset). I am guessing point-based observations are exactly what they sound like (observations taken at specific locations), so perhaps an explanation is unnecessary. But I wanted to raise it, in case (a) it would be preferable to provide a one sentence description and (b) if so, it would be helpful if someone could suggest a suitable sentence (otherwise I can try and come up with one, and seek feedback).

1. I'm not even sure if weatherbench2 allows us to retrieve data from them. I'd just leave it off for now unless we definitely know that it allows us to retrieve data.
2. You can always say something like "point based observations (e.g., weather stations or buoys)"

Steph-Chong commented:
> > There are two areas where I would particularly appreciate feedback:
> > 
> > 1. WeatherBench 2: as I don't have a background in this area, it would be much appreciated if someone could please double check what I have written is accurate and/or if there is anything else that should be mentioned in this section.
> > 2. The "Point-Based Data" subsection: doesn't include a brief description of what point-based observations are. (The other "Datasets" subsections provide a brief description of what type of data is included in the dataset). I am guessing point-based observations are exactly what they sound like (observations taken at specific locations), so perhaps an explanation is unnecessary. But I wanted to raise it, in case (a) it would be preferable to provide a one sentence description and (b) if so, it would be helpful if someone could suggest a suitable sentence (otherwise I can try and come up with one, and seek feedback).
> 
>     1. I'm not even sure if weatherbench2 allows us to retrieve data from them. I'd just leave it off for now unless we definitely know that it allows us to retrieve data.

I was under the impression you could, but perhaps I misunderstood?

https://weatherbench2.readthedocs.io/en/latest/data-guide.html#ground-truth-datasets
https://console.cloud.google.com/storage/browser/weatherbench2
https://weatherbench2.readthedocs.io/en/latest/data-guide.html#forecast-datasets

>     2. You can always say something like "point based observations (e.g., weather stations or buoys)"

Thanks for this suggestion.

The existing sentence says: "Point-based observations are shared routinely between countries for the purposes of weather modelling." 

So perhaps I could modify it to something along the lines of: "Point-based observations (e.g. from weather stations or buoys) are shared routinely between countries for the purposes of weather modelling"?

Steph-Chong commented:
Re. WeatherBench 2 - I'll leave that to everyone else to make a judgement call on. Just let me know once that is resolved so I can update (or not) accordingly.

nicholasloveday commented:
> So perhaps I could modify it to something along the lines of: "Point-based observations (e.g. from weather stations or buoys) are shared routinely between countries for the purposes of weather modelling"?

That sounds good.

And yes, you're right. It does look like weatherbench2 allows you to access data (I haven't tested it out yet). I'm happy with what you've written about weatherbench2

Steph-Chong commented:
I have:
- added some extra info re. point-based observations (thanks @nicholasloveday),
- in the WeatherBench 2 section, added a link to the datasets page in their docs (to make it easier for people who want to quickly locate info regarding their included datasets),
- made a minor edit to remove some repetition in one sentence.

I didn't notice any typos etc. (famous last words...), so if others are happy, I think this is ready now.
